android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java

```diff
@@ -1040,6 +1058,14 @@ public void onBalance(boolean success) {
                                 ? View.GONE
                                 : View.VISIBLE);
                 mWalletBalanceProgress.setVisibility(View.GONE);
+                if (mBraveRewardsNativeWorker.isGrandfatheredUser()
+                        && mBraveRewardsNativeWorker.getExternalWalletType().equals(
+                                BraveWalletProvider.ZEBPAY)
```

Comment:
Would you mind checking `mExternalWallet.getType()` instead? `RewardsServiceImpl::GetExternalWalletType()` was a blunder, and has all sorts of problems...